pkg/e2e/autoscaler/autoscaler.go

```diff
@@ -260,7 +260,6 @@ var _ = g.Describe("[Feature:Machines][Serial] Autoscaler should", func() {
 		g.By(fmt.Sprintf("Creating ClusterAutoscaler configured with maxNodesTotal:%v", maxNodesTotal))
 		clusterAutoscaler := clusterAutoscalerResource(maxNodesTotal)
 		o.Expect(client.Create(context.TODO(), clusterAutoscaler)).Should(o.Succeed())
-		cleanupObjects = append(cleanupObjects, runtime.Object(clusterAutoscaler))
```

Comment:
If you remove this, CA will keep running which may have negative impact on other tests. Instead, we might just dump ca logs right before CA resource is removed.

Reply:
or what about removing the CA at the end of the test, if completed successfully?

Reply:
The test run after the failed test will be potentially impacted if the CA resource is not removed.

Reply:
`cleanupObjects = append(cleanupObjects, runtime.Object(clusterAutoscaler))` still needs to be kept in the test.

Reply:
If we append CA ressource to the cleanupObjects, then again will have to find out CA resource at cleaning up to print the logs. Since CA resource is one special case where we are printing logs, thought of not adding it in cleanupObjects and deleting it specifically after handling cleanupObjects in the defer func. 
Ofcourse, what you suggesting is also another approach. We can add CA to cleanupObjects and then at deletion time , check object type for CA resource. If found, print logs.

I prefered first thinking that its only one resource which needs special handling so may not be good idea to first add and then filter out. WDYT?

Reply:
You already have the CA object assigned to `clusterAutoscaler` variable.

Reply:
> I prefered first thinking that its only one resource which needs special handling so may not be good idea to first add and then filter out. WDYT?

I took this approach: https://github.com/openshift/cluster-api-actuator-pkg/pull/88/commits/eab13a740ef6d5941ca905485118d06017225ef9